server/src/Application/Services/Flashcard/FlashcardFileImporter.cs

```diff
@@ -0,0 +1,63 @@
+using System.Threading;
+using StudyZen.Application.Dtos;
+
+
+namespace StudyZen.Application.Services
+{
+    public class FlashcardFileImporter
+    {
+        private readonly IFlashcardService _flashcardService;
+
+        public FlashcardFileImporter(IFlashcardService flashcardService)
+        {
+            _flashcardService = flashcardService;
+        }
+
+       public void ImportFlashcardsFromCsvStream(Stream stream)
+        {
+            try
+            {
+                using (var reader = new StreamReader(stream))
+                {
+                    var flashcardImportThreads = new List<Thread>();
+
+                    while (!reader.EndOfStream)
+                    {
+                        var line = reader.ReadLine();
+                        if (string.IsNullOrWhiteSpace(line))
+                        {
+                            continue;
+                        }
+
+                        var values = line.Split(',');
+
+                        if (values.Length == 3 && int.TryParse(values[0], out int flashcardSetId))
```

Comment:
The user does not know the id of the flashcard set. I think it would be easier to to just send the flashcards as json in the body. The file should only contain questions and answers